pkg/controller/master/backup/backupTaget.go

```diff
@@ -249,40 +255,46 @@ func validateNFSBackupTarget(destURL string) error {
 	return b.unmount()
 }
 
-func validateS3BackupTarget(target *settings.BackupTarget) error {
-	if target.AccessKeyID == "" || target.SecretAccessKey == "" {
-		return fmt.Errorf("invalid s3 credentials, either accessKeyID or secretAccessKey is empty")
-	}
-
-	var secure bool
-	var endpoint = target.Endpoint
-	if endpoint == "" {
-		endpoint = "s3.amazonaws.com"
-	}
-	if strings.HasPrefix(endpoint, "http://") || strings.HasPrefix(endpoint, "https://") {
-		u, err := url.Parse(endpoint)
-		if err != nil {
-			return err
+func (h *TargetHandler) validateS3BackupTarget(target *settings.BackupTarget) error {
+	os.Setenv("AWS_ACCESS_KEY_ID", target.AccessKeyID)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", target.SecretAccessKey)
+	customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+		if target.Endpoint != "" {
+			return aws.Endpoint{
+				PartitionID: "aws",
+				URL:         target.Endpoint,
+			}, nil
 		}
-		endpoint = u.Host
-		secure = u.Scheme == "https"
-	}
 
-	client, err := minio.NewWithRegion(endpoint, target.AccessKeyID, target.SecretAccessKey, secure, target.BucketRegion)
+		// returning EndpointNotFoundError will allow the service to fallback to it's default resolution
+		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
+	})
+
+	var ca io.Reader
+	if target.Cert != "" {
+		ca = strings.NewReader(target.Cert)
+	}
+	cfg, err := awsconfig.LoadDefaultConfig(h.ctx,
+		awsconfig.WithEndpointResolver(customResolver),
+		awsconfig.WithCustomCABundle(ca),
+		awsconfig.WithDefaultRegion(target.BucketRegion))
 	if err != nil {
 		return err
 	}
 
-	exist, err := client.BucketExists(target.BucketName)
+	// create a s3 service client
+	client := s3.NewFromConfig(cfg)
+	output, err := client.ListBuckets(context.TODO(), &s3.ListBucketsInput{})
```

Comment:
A list operation is unnecessary to check bucket existence. Use HeadBucket()

Reply:
HeadBucket would raise an invalid URL error for other s3 compatible storage services like minio, this is because it will use a combined URL like `https://bucketname.endpoint_path:port`.
```
request send failed, Head \"https://backupbucket.172.16.0.212:31228/\": dial tcp: lookup backupbucket.172.16.0.212: no such host
```